### DISCUSSION TITLE
multithread APBS 3.0 for Win and APBS 3.0 for Mac

### DISCUSSION THREAD

**Discussion by mishaguy** [Electrostatics/apbs#197] [Q&A] [answered]
Hi All,

Does anyone have APBS 3.0 compiled with multithread option on WIndows.
Also does anyone have APBS 3.0 compiled on Mac?
Thanks,
Misha

**Comment by sobolevnrm** [comment]
@intendo or @nsoblath might have a better answer.  Binaries for Mac (and Windows) are available from [GitHub releases](https://github.com/Electrostatics/apbs/releases); however, I do not think they include multi-threaded versions.

**Comment by mishaguy** [comment]
yes, I've checked there and do not see 3.0 for Mac. I use 1.5 but it is a good idea to compile 3.0 because it has so many new things.
yes, on Win exe compiled without multithread.

I just thought to save time if anyone already have these.
THanks

**Comment by sobolevnrm** [comment]
https://github.com/Electrostatics/apbs/releases/download/3.3.0.alpha/APBS-3.3.0.Darwin.zip is the version for Mac.

**Comment by mishaguy** [comment]
I see, got it, thanks

**Comment by intendo** [comment]
We will try to add it to the next release for all platforms.

**Comment by mishaguy** [comment]
for 3.3.0 on Windows, it asks for SuiteSparse.
where to get prebuilt SuiteSparse?

**Reply by intendo** [reply]
You can get it from like we do in Github Actions, https://github.com/Electrostatics/FETK/blob/08667eae76d025924f4bb4d1d395ac55d5def43c/.github/workflows/Build.yaml#L78:   

```wget https://github.com/Electrostatics/cache/releases/download/v0.3.0/apbs_dependencies_vcpkg.zip```

**Comment by mishaguy** [comment]
when do you plan to release normal installer for 3.3.0 on Windows?
this is too much for my users, they need a simple installation steps.

**Reply by intendo** [reply]
The plan is to have a simple apbs-version.zip file that the user can unzip into a directory and be ready to use. A caveat may be that the use has to install any prerequisite software.

**Comment by mishaguy** [comment]
Hi again.
I am back here with the question. Can you compile multithread Windows version of APBS 3.0 and provide it to me (and others)?
I was trying to compile your code and got in so many issues with Cmake, etc.
I thought you know how to do this easily.

Thanks,
Misha

**Comment by intendo** [comment]
I apologize that this has taken so long. This code base is complicated but we are trying to make it easier to build and maintain.
We will be working this week on efforts to make new releases. They will not be multithreaded but that will be on the list of things to get working for the next release in a few months.

**Comment by mishaguy** [comment]
oh, it is very problematic to compile your code on WIndows.
The APBS runs really long without multithreading.
I saw in previous discussions that you provided some users with exe compiled with multithreading.
If you do not have time for this, can you teach how you compile the code on Windows?
I followed Readme and got everything from git, I already edited some places in Cmake to go ahead but still can't produce the build.

I was using such command

cmake -DENABLE_GEOFLOW=ON -DENABLE_BEM=ON -DGET_MSMS=ON -DGET_NanoShaper=ON -DENABLE_PYTHON=ON -G "Visual Studio 17 2022" -A x64 -Dpybind11_DIR=C:\Users\misha\Downloads\apbs\externals\pybind11\tools C:\Users\misha\Downloads\apbs -B build64

I am getting messages like this

CMake Error at tools/python-pybind/CMakeLists.txt:2 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

There is pybind11Config.cmake.in in pybind11 directory.
If I copy it to pybind11Config.cmake, I am getting stocked at

CMake Error at externals/pybind11/tools/pybind11Config.cmake:114:
  Parse error.  Expected a command name, got unquoted argument with text
  "@PACKAGE_INIT@".

What is the way to go around this? I am not exactly IT professional and would appreciate your help.
Thanks,
Misha

**Comment by intendo** [comment, marked as answer]
The best we can do right now is point you to our [latest branch](https://github.com/Electrostatics/apbs/tree/nsoblath/issue_163)  attempt towards getting a new release. You can look at the [Github Actions file] (https://github.com/Electrostatics/apbs/blob/nsoblath/issue_163/.github/workflows/build.yml) to see the Windows environment and build commands needed to build APBS on Windows. Unfortunately, we don't have more resources to address all the activities we would like to complete. If you are able to get APBS compiled on Windows, please let us know how you were able to do it so we can share it with others.
